pkg/operation/botanist/etcd_test.go

```diff
@@ -0,0 +1,500 @@
+		kubernetesClient = mockkubernetes.NewMockInterface(ctrl)
+		c = mockclient.NewMockClient(ctrl)
+		botanist = &Botanist{Operation: &operation.Operation{}}
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	Describe("#DefaultEtcd", func() {
+		var hvpaEnabled = true
+
+		BeforeEach(func() {
+			botanist.K8sSeedClient = kubernetesClient
+			botanist.Seed = &seedpkg.Seed{
+				Info: &gardencorev1beta1.Seed{},
+			}
+			botanist.Shoot = &shootpkg.Shoot{
+				Info: &gardencorev1beta1.Shoot{
+					Spec: gardencorev1beta1.ShootSpec{
+						Maintenance: &gardencorev1beta1.Maintenance{
+							TimeWindow: &maintenanceTimeWindow,
+						},
+					},
+				},
+				SeedNamespace: namespace,
+			}
+		})
+
+		Context("no shooted seed", func() {
+			BeforeEach(func() {
+				botanist.ShootedSeed = nil
+			})
+
+			It("should successfully create an etcd interface (normal class)", func() {
+				defer test.WithFeatureGate(gardenletfeatures.FeatureGate, features.HVPA, hvpaEnabled)()
+				kubernetesClient.EXPECT().Client()
+
+				validator := &newEtcdValidator{
+					expectedClient:                  BeNil(),
+					expectedNamespace:               Equal(namespace),
+					expectedRole:                    Equal(role),
+					expectedClass:                   Equal(class),
+					expectedRetainReplicas:          BeFalse(),
+					expectedStorageCapacity:         Equal("10Gi"),
+					expectedDefragmentationSchedule: Equal(pointer.StringPtr("34 12 */3 * *")),
+					expectedHVPAConfig: Equal(&etcd.HVPAConfig{
+						Enabled:               hvpaEnabled,
+						MaintenanceTimeWindow: maintenanceTimeWindow,
+					}),
+				}
+
+				oldNewEtcd := NewEtcd
+				defer func() { NewEtcd = oldNewEtcd }()
+				NewEtcd = validator.NewEtcd
+
+				etcd, err := botanist.DefaultEtcd(role, class)
+				Expect(etcd).NotTo(BeNil())
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("should successfully create an etcd interface (important class)", func() {
+				class := etcd.ClassImportant
+
+				defer test.WithFeatureGate(gardenletfeatures.FeatureGate, features.HVPA, hvpaEnabled)()
+				kubernetesClient.EXPECT().Client()
+
+				validator := &newEtcdValidator{
+					expectedClient:                  BeNil(),
+					expectedNamespace:               Equal(namespace),
+					expectedRole:                    Equal(role),
+					expectedClass:                   Equal(class),
+					expectedRetainReplicas:          BeFalse(),
+					expectedStorageCapacity:         Equal("10Gi"),
+					expectedDefragmentationSchedule: Equal(pointer.StringPtr("34 12 */3 * *")),
+					expectedHVPAConfig: Equal(&etcd.HVPAConfig{
+						Enabled:               hvpaEnabled,
+						MaintenanceTimeWindow: maintenanceTimeWindow,
+					}),
+				}
+
+				oldNewEtcd := NewEtcd
+				defer func() { NewEtcd = oldNewEtcd }()
+				NewEtcd = validator.NewEtcd
+
+				etcd, err := botanist.DefaultEtcd(role, class)
+				Expect(etcd).NotTo(BeNil())
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+
+		Context("no shooted seed", func() {
```

Comment:
There is already a context with the same name in L100.

```suggestion
		Context("for shooted seed", func() {
```